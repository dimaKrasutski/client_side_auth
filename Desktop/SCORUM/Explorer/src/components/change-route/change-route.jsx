import React from 'react';
import { connect } from 'react-redux';
import { ChangeRoute as ChangeRouteUI } from 'theme/components/change-route/change-route';

const mapStateToProps = state => ({
  locale: state.intl.localeFull,
});

export const ChangeRoute = connect(mapStateToProps)(ChangeRouteUI);

ChangeRoute.to = to => () => <ChangeRoute to={to} />;
