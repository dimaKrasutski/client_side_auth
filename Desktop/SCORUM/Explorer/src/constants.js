import * as THEME_CONSTANTS from 'theme/constants';

export const {
  LOCALE_KEYS,
  LANG_KEYS,
  LOCALE,
  LOCALES,
  DEFAULT_LOCALE,
  DEFAULT_LANG,
} = THEME_CONSTANTS;

export const UNDEFINED_ERROR_ID = 'explorer.error.unknown';

export const MIN_FRACTION_DIGITS = 3;
export const MAX_FRACTION_DIGITS = 9;

export const CURRENCY_USD = 'USD';
export const CURRENCY_EUR = 'EUR';
export const ALL_CURRENCY = [CURRENCY_USD, CURRENCY_EUR];
