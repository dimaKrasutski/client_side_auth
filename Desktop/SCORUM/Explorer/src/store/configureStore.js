import { createStore, applyMiddleware, compose } from 'redux';
import { createEpicMiddleware } from 'redux-observable';
import { rootEpic } from '../epics/root';
import rootReducer from '../reducers/index';

const epicMiddleware = createEpicMiddleware(rootEpic);

export default function configureStore(initialState) {
  const middlewares = [
    epicMiddleware,
  ];
  if (process.env.PROJECT_ENV !== 'production') {
    // eslint-disable-next-line
    const { logger } = require(`redux-logger`);
    middlewares.push(logger);
  }

  const enhancer = compose(applyMiddleware(...middlewares));

  return createStore(rootReducer, initialState, enhancer);
}
