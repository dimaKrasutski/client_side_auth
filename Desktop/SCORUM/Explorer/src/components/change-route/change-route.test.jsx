import configureStore from 'redux-mock-store';
import { ChangeRoute } from './change-route';
import { DEFAULT_LOCALE } from '../../constants';

const mockStore = configureStore({});
const store = mockStore({
  intl: {
    localeFull: DEFAULT_LOCALE,
  },
});

describe('<RouteLink />', () => {
  test('should use state + locale', () => {
    const wrapper = shallow(<ChangeRoute to="/test" store={store}>Link here!</ChangeRoute>).dive();
    expect(wrapper).toMatchSnapshot();
  });

  describe('when static to is called', () => {
    test('should create component and render', () => {
      const Component = ChangeRoute.to('/another/');
      const wrapper = shallow(<Component store={store} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
