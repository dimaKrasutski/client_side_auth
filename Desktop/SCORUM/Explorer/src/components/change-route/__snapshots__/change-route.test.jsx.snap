// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RouteLink /> should use state + locale 1`] = `
<Redirect
  push={false}
  to={
    Object {
      "pathname": "/en-us/test",
      "search": "",
    }
  }
/>
`;

exports[`<RouteLink /> when static to is called should create component and render 1`] = `
<Connect(ChangeRoute)
  to="/another/"
/>
`;
